src/content/docs/reference/driver-guide.md

```diff
@@ -0,0 +1,452 @@
+
+```js
+// index.ts
+export const EmailCollection = {
+  // Get a single email by ID
+  async one({ id }: { id: string }) {
+    const data = await api("GET", `emails/${id}`);
+    return { ...data };
+  },
+
+  // Send a new email
+  async send(args: {
+    from: string;
+    to: string[];
+    subject: string;
+    html?: string;
+    text?: string;
+  }) {
+    const data = await api("POST", "emails", undefined, args);
+    return { ...data };
+  }
+};
+
+```
+
+### Email Resource
+
+Resources handle operations specific to a single item and must implement a `gref`  for referencing:
+
+```js
+// index.ts
+export const Email = {
+  // Create a reference to this email
+  gref: function (_, { obj }) {
+    return root.emails.one({ id: obj.id });
+  },
+
+  // Update email (e.g., reschedule)
+  async update(args: { scheduled_at: string }, { self }) {
+    const { id } = self.$argsAt(root.emails.one);
+    return api("PATCH", `emails/${id}`, undefined, args);
+  },
+
+  // Cancel a scheduled email
+  async cancel(_, { self }) {
+    const { id } = self.$argsAt(root.emails.one);
+    return api("POST", `emails/${id}/cancel`);
+  }
+};
+
+```
+
+### Domain Collection (Pagination Example)
+
+For resources that support listing, collections should implement a `page()` function for pagination:
+
+```js
+// index.ts
+export const DomainCollection = {
+  // Get a single domain
+  async one({ id }: { id: string }) {
+    return await api("GET", `domains/${id}`);
+  },
+
+  // List domains with pagination
+  async page() {
+    const result = await api("GET", "domains");
+    return {
+      items: result.data
+    };
+  },
+
+  // Create a new domain
+  async create(args: { name: string; region?: string }) {
+    return await api("POST", "domains", undefined, args);
+  }
+};
+```
+
+### Graph References (grefs)
+
+In Membrane, grefs (graph references) create a way to reference and track specific resources across your graph. Every resource should implement a `gref` function that returns a unique, consistent path to that resource.
+
+Think of a gref like a URL - it's a way to point to a specific resource that can be stored, passed around, and used later. For example:
+
+```js
+// In Email resource
+gref: function (_, { obj }) {
+  return root.emails.one({ id: obj.id });
+}
+
+// This creates references like:
+// resend:emails.one(id:"12345")
+```
+
+Grefs serve several important purposes:
+
+1. **Resource Identity**: They provide a standardized way to reference specific resources
+2. **Resource Lookups**: Programs can use grefs to consistently find and access resources
+3. **Cross-Program Communication**: Programs can pass grefs to each other to reference the same resource
+4. **Action Context**: When performing actions on a resource, the gref provides context about which resource to act on using `self.$argsAt()`
+
+### Testing Your Driver
+
+Create a separate `tests.ts` file to organize your tests. Here's an example from the Resend driver:
```

Comment:
I wonder if it's worth noting that tests are somewhat of a bonus points thing to implement. That might be obvious, though. Also maybe that you can run tests on a cron.